src/metabase/sync/analyze/query_results.clj

```diff
@@ -40,14 +40,10 @@
   special type with a nil special type"
   [result-metadata col]
   (update result-metadata :special_type (fn [original-value]
-                                          ;; If the original special type is a PK or FK, we don't want to use a new
-                                          ;; computed special type because it'll just be confusing as we can't do any
-                                          ;; meaningful binning etc on it. If it's not of that type and we are able to
-                                          ;; compute a special type based on the results, use that
-                                          (if-let [new-special-type (and (not (isa? original-value :type/PK))
-                                                                         (not (isa? original-value :type/FK))
-                                                                         (classify-name/infer-special-type col))]
-                                            new-special-type
+                                          ;; If we already know the special type, becouse it is stored, don't classify again.
+                                          ;; Also try to refine special type set upstream for aggregation cols (which comes back as :type/Number)
+                                          (case original-value
+                                            (nil :type/Number) (classify-name/infer-special-type col)
```

Comment:
I'm a little confused here. First, why do we only want to rerun this code if there is no special type, or it's `=` a `:type/Number`. Seems like you'd want that to be `isa?` `:type/Number` so that we don't miss integers, floats, etc. Second, what's the rationale/trade-offs around recalculating that for numbers but not anything else. Would there be value in scanning the results to figure out that this just isn't `:type/Text` but these query results are email addresses (i.e. `type/Email`). Is the cost of that scanning pretty high, which is why it's not worth it?

Reply:
The :type/Number thing is a special case. For some reason we tag aggregate fields as such -- and it's always type/Number -- in qp/annotate (breaking the convention that special type should be used for types with semantic meaning). It's a wart I indent to get to the bottom of, but I think that's out of scope for this PR.